enumerables.rb

```diff
@@ -0,0 +1,203 @@
+              to_a
+            else
+              self
+            end
+      (0..arr.length - 1).each do |index|
+        yield(arr[index], index)
+      end
+      self
+    else
+      to_enum :my_each_with_index
+    end
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    new_a = []
+    my_each do |element|
+      new_a << element if yield(element)
+    end
+    new_a
+  end
+
+  # My_all?
+  def my_all?(arg = nil)
+    arr = to_a if self.class == Range
+    arr = self if self.class != Range
+    if block_given?
+      arr.length.times do |i|
+        return false if yield(arr[i]) == false
+      end
+    else
+      puts 'block was not given'
+      puts "self class is #{arr.class}"
+      puts "argument class is #{arg.class}"
+      if !arg.nil?
+        arr.length.times do |i|
+          if arg.class == Integer
+            unless arr[i].to_s.include?(arg.to_s)
+              puts(" array item #{arr[i]} doesn't equal argument #{arg}, rendering false")
+              return false
+            end
+          elsif arg.class == Class
+            unless arr[i].class == arg
+              puts(" array item #{arr[i]} doesn't equal argument #{arg}, rendering false")
+              return false
+            end
+          else
+            unless arr[i].to_s.include?(arg.source)
+              puts(" array item #{arr[i]} doesn't equal argument #{arg.source}, rendering false")
+              return false
+            end
+          end
+        end
+      else
+        arr.length.times do |i|
+          if arr[i] == false || arr[i].nil?
+            puts arr[i]
+            return false
+          end
+        end
+      end
+    end
+    true
+  end
+
+  # my_any?
+  def my_any?(parameter = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) }
+      false
+    elsif parameter.nil?
+      my_each { |n| return true if n }
+    elsif !parameter.nil? && (parameter.is_a? Class)
+      my_each { |n| return true if [n.class, n.class.superclass].include?(parameter) }
+    elsif !parameter.nil? && parameter.class == Regexp
+      my_each { |n| return true if parameter.match(n) }
+    else
+      my_each { |n| return true if n == parameter }
+    end
+    false
+  end
+
+  # none?
+  def my_none?(arg = nil)
+    arr = to_a if self.class == Range
+    arr = self if self.class != Range
+    if block_given?
+      arr.length.times do |i|
+        return false if yield(arr[i]) == true
+      end
+    elsif !arg.nil?
+      arr.length.times do |i|
+        if arg.class == Integer
+          return false if arr[i].to_s.include?(arg.to_s)
+        elsif arg.class == Class
+          return true unless arr[i].class == arg
+        else
+          return true unless arr[i].to_s.include?(arg.source)
+        end
+      end
+    else
+      puts 'Argument not given'
+      arr.length.times do |i|
+        return false unless arr[i] == false || arr[i].nil?
+      end
+    end
+    true
+  end
```

Comment:
Please make sure your `my_none` correctly handles regular expressions at it's input, as to avoid the following error:

```
  3) enumerables #my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
     Failure/Error: expect(words.my_none?(/d/)).to be words.none?(/d/)
     
       expected false
            got true
     # ./spec/enumerables_spec.rb:325:in `block (4 levels) in <top (required)>'
```

Try using it like this to get the correct result:

```
# This should return false
%w[dog cat monkey].my_none?(/d/)
```